### ISSUE TITLE
The "play over desktop" feature may capture an erased screen when started naturally

### ISSUE THREAD

**Issue by pierce-smith1** [pierce-smith1/yok.scr#37]
When yok.scr has the "play over desktop" option checked, it grabs a screenshot of the current desktop and stores that for the lifetime of the program. However, when started naturally by the screensaver timeout, some computers seem to blank the screen before actually starting the screensaver, causing the desktop to never get captured. Ugh.

Some possible fixes:
* Figure out what window is blanking the screen after the screensaver timeout, then just don't capture that window?
* Somehow get a screenshot through some other means..?
* Find out how to configure around the problem and tell all our users to do that / do it automatically?

**Comment by pierce-smith1** [comment]
It turns out this is a security feature of Windows and we cannot get around it. "Secure" screensavers launch in a completely separate desktop environment that cannot access information from the user desktop environment. Interestingly, screensavers are only supposed to be "secure" if you check the "require password when waking up" option, but all of my test computers and plenty of other people's computers are doing it anyway. Once Windows decides to launch the screensaver securely, there's nothing we can do, we simply can't get that screenshot.

https://learn.microsoft.com/en-us/windows/win32/winstation/desktops

What makes this extra awkward is that, from what I can tell, not everyone's computers launch the screensaver securely, so this option will work for some people and not for others. This means I don't want to just _get rid of it_, but it simply isn't practical to make it work for everyone, and I'm almost certain that those it works for will stop working eventually. So, what the hell do we do?

**Comment by McAwesome123** [comment]
So far the only options to fix this seem to be:
- Use an older version of Windows that doesn't have this feature (like Windows 7)... if it even works on those versions.
- Try to work around this feature externally (as the screensaver by itself is very unlikely to be able to do so).

Some examples of workarounds include:
- Disabling the screensaver in Windows and instead using task scheduler or another program to start the screensaver. (Potential issues: Janky as hell; Inconsistent behaviour with default screensavers)
- Using task scheduler or another program to kill and rerun the screensaver when it starts. (Potential issues: Also janky as hell; Requires complicated setup and/or trust)

To put it simply: There are no good solutions

**Comment by pierce-smith1** [comment]
> Disabling the screensaver in Windows and instead using task scheduler or another program to start the screensaver

> Using task scheduler or another program to kill and rerun the screensaver when it starts

To be specific on the jank aspect, I think these solutions are not befitting of this project because:
* Require user intervention just to use your damn screensaver
* May require having two copies of yokscr, meaning you have to update them both in parallel (and you aren't going to remember to do that)
* Are brittle (what happens when windows randomly decides it doesn't like your scheduled tasks or your group policy settings and deletes them? It's happened before!)
* Are ridiculously overcomplicated, and will result in more effort than is worth to support

Closing because we will not fix this.
